code_doc_gen/ai_providers/base.py

```
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Dict, Any, List
+import logging
+
+class AIProvider(ABC):
+    """Abstract base class for AI providers."""
+    
+    def __init__(self, api_key: str, model: str = None):
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)